### PR TITLE
Fix for read stream being too quick for MySQLStream in ImportFeedCommand

### PR DESCRIPTION
**Pull request by pies**

### CODE REVIEW

File: src/cli/ImportFeedCommand.ts [hunk 1]
```diff
@@ -1,35 +1,30 @@
...
 const getExt = filename => path.extname(filename).slice(1).toUpperCase();
-const readFile = filename => byline.createStream(fs.createReadStream(filename, "utf8"));
 
 /**
  * Imports one of the feeds
  */
 export class ImportFeedCommand implements CLICommand {
 
-  private index: {[name: string]: MySQLTable} = {};
+  private index: { [name: string]: MySQLTable } = {};
```
Comment (piotrjura) [inline src/cli/ImportFeedCommand.ts]: You can use Record<Key, Type> here:
`private index: Record<string, MySQLTable>;`

Comment (piotrjura) [reply src/cli/ImportFeedCommand.ts]: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype

File: src/cli/ImportFeedCommand.ts [hunk 2]
```diff
@@ -1,35 +1,30 @@
...
     protected readonly db: DatabaseConnection,
     protected readonly files: FeedConfig,
     protected readonly tmpFolder: string,
     private readonly xFilesFolder?: string,
-  ) { }
+  ) {
+  }
 
   protected get fileArray(): FeedFile[] {
     return Object.values(this.files);
```
Comment (piotrjura) [inline src/cli/ImportFeedCommand.ts:30]: As far as I can see, this.files is readonly.

I'm not sure where fileArray() is used, but it should be computed or something, as it basically operates on constant.

File: src/cli/ImportFeedCommand.ts [hunk 3]
```diff
@@ -159,10 +153,40 @@ export class ImportFeedCommand implements CLICommand {
...
-    const stream = readFile(path + filename).pipe(tableStream);
 
     try {
-      await streamToPromise(stream);
+      await new Promise((resolve, reject) => {
+        const lineStream = new LineByLineReader(path + filename);
+
+        lineStream.on('line', async function (line: any) {
+          // Pause after every line to enforce correct line ordering in multi-record files.
+          lineStream.pause();
```
Comment (piotrjura) [inline src/cli/ImportFeedCommand.ts:163]: Don't use `any` if the type is known.
line is of type Buffer.

Also not sure how many "lines" we're talking about here. You can also just accumulate lines into Buffer, and then let's say every 1000 lines or so write to the writable buffer.

Pausing the stream might cause the data to accumulate into the buffer anyway (thus using a lot of memory). Haven't tried that, but read about how it works though.

**Review by piotrjura — APPROVED** [review]

File: src/cli/ImportFeedCommand.ts (hunk 3 above)
Comment (pies) [reply src/cli/ImportFeedCommand.ts:163]: It's fine, the code in line-by-line handles that correctly, and I'd rather have it handle the abstraction. It does its own abstraction of reading X bytes from a file, splitting into lines and feeding me line by line, so I don't think there's a need for an extra buffer.

File: src/cli/ImportFeedCommand.ts (hunk 2 above)
Comment (pies) [reply src/cli/ImportFeedCommand.ts:30]: It's a small object and this doesn't happen often enough to be an optimization target.

**Pull request merged**